pkg/trace/writer/trace_test.go

```diff
@@ -56,14 +65,22 @@ func TestTraceWriter(t *testing.T) {
 			"Content-Encoding":             "gzip",
 		}
 
-		// Ensure that the number of payloads and their contents match our expectations. The MaxSpansPerPayload we
-		// set to 4 at the beginning should have been respected whenever possible.
+		// we should have 4 payloads based on the configured flush threshold for this test.
 		assert.Len(testEndpoint.SuccessPayloads(), 4, "We expected 4 different payloads")
 		assertPayloads(assert, traceWriter, expectedHeaders, sampledTraces, testEndpoint.SuccessPayloads())
 	})
 }
 
-func calculateTracePayloadSize(sampledTraces []*TracePackage) int64 {
+func calculateTracePayloadEstimatedSize(sampledTraces []*TracePackage) int {
+	var size int
+	for _, pkg := range sampledTraces {
+		size += pkg.Trace.Msgsize()
+		size += pb.Trace(pkg.Events).Msgsize()
```

Comment:
This can be updated to use `pkg.size()` now

Reply:
Done